aioshelly/rpc_device/wsrpc.py

```diff
@@ -357,52 +374,129 @@ async def call(
+                if TYPE_CHECKING:
+                    # _raise_for_unrecoverable_errors ensures that auth_data is not None
+                    assert self._auth_data is not None
+                auth = json_loads(resp["error"]["message"])
+                self._session.auth = self._auth_data.get_auth(
+                    auth["nonce"], auth.get("nc", 1)
+                )
+            to_retry.append(call)
+
+        _, results = await self._rpc_calls(
+            [(call.method, call.params) for call in to_retry], timeout
+        )
+        for call in results:
+            if (result := call.result) is None:
+                resp = call.resolve.result()
+                self._raise_for_unrecoverable_errors(resp, allow_auth_retry=False)
+            else:
+                successful.append(result)
+
+        return successful
+
+    async def _rpc_calls(
+        self, rpc_calls: Iterable[tuple[str, dict[str, Any] | None]], timeout: int
+    ) -> tuple[bool, list[RPCCall]]:
+        """Websocket RPC call.
+
+        calls is a tuple of tuples of
+        (
+           (method, params),
+           ...
+
+        """
         if self._client is None:
             raise RuntimeError("Not connected")
 
-        future: asyncio.Future[dict[str, Any]] = self._loop.create_future()
-        call = RPCCall(self._next_id, method, params, self._session, future)
-        self._calls[call.call_id] = call
+        sent_calls: list[RPCCall] = []
+        loop = self._loop
+        all_successful: bool = True
+        future: asyncio.Future[dict[str, Any]]
 
         try:
             async with asyncio.timeout(timeout):
-                await self._send_json(call.request_frame)
-                resp = await future
+                for method, params in rpc_calls:
+                    call_id = self._next_id
+                    future = loop.create_future()
+                    call = RPCCall(call_id, method, params, self._session, future)
+                    sent_calls.append(call)
+                    self._calls[call_id] = call
+                    await self._send_json(call.request_frame)
+
+                # Wait for all the responses
+                for call in sent_calls:
+                    response = await call.resolve
+                    if "result" not in response:
+                        all_successful = False
+                        continue
+                    call.result = response["result"]
         except TimeoutError as exc:
-            with contextlib.suppress(asyncio.CancelledError):
-                future.cancel()
-                await future
+            for call in sent_calls:
+                with contextlib.suppress(asyncio.CancelledError):
+                    call.resolve.cancel()
+                    await call.resolve
+                # Ensure the call is removed from the calls dict
+                # on failure
+                self._calls.pop(call.call_id, None)
```

Comment:
This is technically a bug fix but its probably not a real problem since if we have timeouts it generally means the connection is going to get torn down anyways and the dict will be cleared